notebooks/tetraopt_example_using_comms_channel.ipynb

```diff
@@ -0,0 +1,162 @@
+    "            algorithm=AlgorithmProto.TETRA_OPT,\n",
+    "            experiment_id=exp_sample.id,\n",
+    "            compute=HardwareProto.SMALL,\n",
+    "            parameters={'parameters': tetra_opt_params, 'inputs': {}}\n",
+    "        )\n",
+    "\n",
+    "        print(f\"Experiment run id: {run.id}\")\n",
+    "\n",
+    "        #The \"ask\" argument is provided by tetraopt, it is an array with a dimension of n(ask.headers)\n",
+    "        async def objective_function_callback(ask: Ask) -> Tell:    \n",
+    "            dim = len(ask.headers)\n",
+    "            \n",
+    "            #initialize Ackley function to receive n dimension arrays\n",
+    "            func = otf.Ackley(dim)\n",
+    "            y = []\n",
+    "\n",
+    "            \n",
+    "            #append the results of the Ackley function to y for all given n dimension parameters\n",
+    "            for parameter in ask.parameters:\n",
+    "                y.append(float(func(np.array(parameter.values))))\n",
+    "\n",
+    "            #create and return a Tell object which has the parameters and the results mapped together\n",
+    "            tell = Tell(\n",
+    "                parameters=ask.parameters,\n",
+    "                headers=ask.headers,\n",
+    "                results=y\n",
+    "            )\n",
+    "            return tell\n",
+    "\n",
+    "        \n",
+    "        async def local_optimization_function_callback(ask: Ask) -> Tell:\n",
+    "            dim = len(ask.headers)\n",
+    "            func = otf.Ackley(dim)\n",
+    "            y = []\n",
+    "            new_x = []\n",
+    "            for parameter in ask.parameters:\n",
+    "                res = optimize.minimize(func, np.array(parameter.values))\n",
+    "                new_x.append({\"values\": res.x})\n",
+    "                y.append(float(res.fun))\n",
+    "           \n",
+    "            tell = Tell(\n",
+    "                parameters=ask.parameters,\n",
+    "                headers=ask.headers,\n",
+    "                results=y,\n",
+    "                candidates=new_x\n",
+    "            )\n",
+    "\n",
+    "            return tell\n",
+    "        \n",
+    "        def success():\n",
+    "            print(\"One async function done!\")\n",
+    "\n",
+    "        await asyncio.gather(\n",
+    "            objective_func_channel.connect(\n",
+    "                 callback=objective_function_callback, finish_callback=success, max_duration_in_sec=None, message_timeout_in_sec=500\n",
+    "            ),\n",
+    "            local_opt_channel.connect(\n",
+    "                callback=local_optimization_function_callback, finish_callback=success, max_duration_in_sec=None, message_timeout_in_sec=500\n",
+    "            )\n",
+    "        )\n",
+    "\n",
+    "        print(f\"Experiment run result: {run.check()}\")\n",
+    "        return run\n",
+    "\n",
+    "run = await main()\n",
+    "print(run.poll())"
```

Comment:
Did you just literally posted all of the code into one single cell? Jesus christ 🤣 ? What is then the difference between jupyter notebook and a script???